ytfzf

```diff
@@ -345,18 +347,13 @@ preview_img () {
 download_thumbnails () {
        #scrapes the urls of the thumbnails of the videos from the adjusted json
        thumb_urls="$(printf "%s" "$*" |\
-               jq  -r '.[]|(.thumbs,.videoID)' )"
-	i=0
+               jq  -r '.[]|.thumbs' )"
 	while read -r line; do
-		if [ $((i % 2)) -eq 0 ]; then
-			url="$(echo "$line" | sed -E 's/^"//;s/\.png.*/\.png/')" 
-		else
-			name="$line"
-			{ 
-				curl -s "$url" > "$thumb_dir/$name.png" 
-			} &
-		fi
-		i=$((i + 1))
+		url="$line"
+		name="$(echo "$url" | grep -Po 'vi/\K.*(?=/)')"
+		{ 
+			curl -s "$url" > "$thumb_dir/$name.png" 
+		} &
```

Comment:
This wouldn't impact the time. Since jq is scraping all together in the start.  
The major time improvements would come from the position of `wait` and `curl`s

In this method, we are relying on the videoID to be in the thumbnail. ( wouldn't work well with other sites).